### PR TITLE
advanced_flask homework done

### PR DESCRIPTION
**Pull request by oleksii-manzik**

### CODE REVIEW

File: flask3/products_blueprint/main.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        selected_products = [(product["id"], product["name"])
+                             for product in data]
+    return render_template('all_products.html',
+                           selected_products=selected_products)
+
+
+@products.route('/product/<product_id>')
+def get_product_page(product_id):
+    session[product_id] = 'clicked'
+    data = get_data('products_blueprint/product_data.json')
```
Comment (dimigor) [inline flask3/products_blueprint/main.py]: you have constant for this path

File: flask3/utils.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import json
+
+
+def get_data(json_file):
+    try:
+        with open(json_file, 'r') as file:
+            return json.load(file)
+    except (FileNotFoundError, json.decoder.JSONDecodeError):
+        return []
```
Comment (dimigor) [inline flask3/utils.py]: return empty dict, not list

File: flask3/utils.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import json
+
+
+def get_data(json_file):
+    try:
+        with open(json_file, 'r') as file:
```
Comment (dimigor) [inline flask3/utils.py]: `file` is reserved python word, name variables better

**Review by dimigor — CHANGES_REQUESTED** [review]

File: flask3/products_blueprint/main.py (hunk 1 above)
Comment (oleksii-manzik) [reply flask3/products_blueprint/main.py]: fixed

File: flask3/products_blueprint/templates/product.html [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{% extends "base.html" %}
+{% block main_content %}
+<h1>{{ product_data.name }}</h1>
```
Comment (dimigor) [inline flask3/products_blueprint/templates/product.html:3]: tabulation

**Review by dimigor — APPROVED** [review]

**Pull request merged**